cmake/traccc-vecmem.cmake

```diff
@@ -1,40 +0,0 @@
-# TRACCC library, part of the ACTS project (R&D line)
-#
-# (c) 2021 CERN for the benefit of the ACTS project
-#
-# Mozilla Public License Version 2.0
-
-# Guard against multiple includes.
-include_guard( GLOBAL )
```

Comment:
I'm not really in favour of this... I would much rather change this line (and the equivalent line in detray/algebra-plugins) to:

```cmake
if( TARGET "vecmem::core" )
   return()
endif()
```

, and keep all of them in place. So that whoever is called first, wins.

Note that this can actually be absolutely necessary in case we don't want to build VecMem as part of this project, but rather pick up a pre-built version of it. Since remember, `find_package(vecmem)` does not declare the VecMem library targets globally! It only declares them in the scope in which `find_package(vecmem)` was called. (We had to realise this in https://github.com/acts-project/acts/pull/998 with @konradkusiak97.) So one of those submodules setting up VecMem for itself through `find_package(vecmem)` will not make the libraries available in this main project.